doxia-integration-tools/src/main/java/org/apache/maven/doxia/tools/DefaultSiteTool.java

```diff
@@ -911,6 +924,29 @@ private File resolveSiteDescriptor(
         return siteDescriptor;
     }
 
+    // TODO Remove this transient method when everyone has migrated to Maven Site Plugin 4.0.0+
+    private void deletePseudoSiteDescriptorMarkerFile(RepositorySystemSession repoSession, ArtifactRequest request) {
+        LocalRepositoryManager lrm = repoSession.getLocalRepositoryManager();
+
+        LocalArtifactRequest localRequest = new LocalArtifactRequest();
+        localRequest.setArtifact(request.getArtifact());
+
+        LocalArtifactResult localResult = lrm.find(repoSession, localRequest);
+        File localArtifactFile = localResult.getFile();
+
+        try {
+            if (localResult.isAvailable() && Files.size(localArtifactFile.toPath()) == 0L) {
+                LOGGER.debug(
+                        "Deleting 0-byte pseudo marker file for artifact '{}' at '{}'",
+                        localRequest.getArtifact(),
+                        localArtifactFile);
+                Files.delete(localArtifactFile.toPath());
+            }
+        } catch (IOException e) {
+            LOGGER.debug("Failed to delete 0-byte pseudo marker file for artifact '{}'", localRequest.getArtifact());
```

Comment:
Shouldn't you include the exception itself as well? Or maybe that's not the convention, I don't know. Anyway, some logging is better than no logging!

Reply:
Stupid me, it is late and  I am not fresh anymore ;-)

Reply:
No problem. Looks like you've added the exception so I'm all set.